cloudformation/aws-parallelcluster.cfn.json

```diff
@@ -1760,12 +1760,70 @@
               "Action": [
                 "ec2:DescribeInstances",
                 "ec2:DescribeLaunchTemplates",
-                "ec2:RunInstances",
                 "ec2:DescribeInstanceStatus",
                 "ec2:CreateTags"
               ],
               "Resource": "*"
             },
+            {
+              "Sid": "EC2RunInstances",
+              "Effect": "Allow",
+              "Action": [
+                "ec2:RunInstances"
+              ],
+              "Resource": [
+                {
+                  "Fn::Sub": [
+                    "arn:${AWS::Partition}:ec2:${AWS::Region}:${AWS::AccountId}:subnet/${ComputeSubnetId}",
+                    {
+                      "ComputeSubnetId": {
+                        "Fn::If": [
+                          "UseMasterSubnetForCompute",
+                          {
+                            "Ref": "MasterSubnetId"
+                          },
+                          {
+                            "Fn::If": [
+                              "CreateComputeSubnetForCompute",
+                              {
+                                "Ref": "ComputeSubnet"
+                              },
+                              {
+                                "Ref": "ComputeSubnetId"
+                              }
+                            ]
+                          }
+                        ]
+                      }
+                    }
+                  ]
+                },
+                {
+                  "Fn::Sub": "arn:${AWS::Partition}:ec2:${AWS::Region}:${AWS::AccountId}:network-interface/*"
+                },
+                {
+                  "Fn::Sub": "arn:${AWS::Partition}:ec2:${AWS::Region}:${AWS::AccountId}:instance/*"
+                },
+                {
+                  "Fn::Sub": "arn:${AWS::Partition}:ec2:${AWS::Region}:${AWS::AccountId}:volume/*"
+                },
+                {
+                  "Fn::Sub": "arn:${AWS::Partition}:ec2:${AWS::Region}::image/ami-*"
+                },
+                {
+                  "Fn::Sub": "arn:${AWS::Partition}:ec2:${AWS::Region}:${AWS::AccountId}:key-pair/*"
+                },
+                {
+                  "Fn::Sub": "arn:${AWS::Partition}:ec2:${AWS::Region}:${AWS::AccountId}:security-group/*"
+                },
+                {
+                  "Fn::Sub": "arn:${AWS::Partition}:ec2:${AWS::Region}:${AWS::AccountId}:launch-template/*"
```

Comment:
Since launch templates are named "${ClusterName}-{{ queue }}-{{ instance_type }}" we could use ${ClusterName}-* to narrow this permission even further

Reply:
I tried, but it didn't work. Probably the Resource arn requires a launch template id instead of the name.